### ISSUE TITLE
Tensorflow 2.1 Jacobian fails when using Keras LeakyReLU Layer

### ISSUE THREAD

**Issue by cianciosa** [tensorflow/tensorflow#36256]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes. See attached test case.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac OS 10.14.6
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: Unsure
- TensorFlow installed from (source or binary): Binary installed from pip
- TensorFlow version (use command below): ('v2.1.0-rc2-17-ge5bf8de410', '2.1.0')
- Python version: Python 2.7.15 and Python 3.7.0
- Bazel version (if compiling from source): N/A
- GCC/Compiler version (if compiling from source): N/A
- CUDA/cuDNN version: N/A
- GPU model and memory: N/A

**Describe the current behavior**
Computing the Jacobian using the Gradient tape function fails when a the LeakyReLU Keras layer is used.

**Describe the expected behavior**
It was expected to calculate the Jacobian of the model with respect to the trainable parameters.

**Code to reproduce the issue**
```
import tensorflow

model = tensorflow.keras.models.Sequential()
model.add(tensorflow.keras.layers.Dense(2,
                                        input_dim=1,
                                        use_bias=True))
model.add(tensorflow.keras.layers.LeakyReLU())
#model.add(tensorflow.keras.layers.ReLU())
model.add(tensorflow.keras.layers.Dense(1,
                                        use_bias=True))

inputs = tensorflow.random.uniform((100,1), 1.0, -1.0)

with tensorflow.GradientTape(watch_accessed_variables=False) as tape:
	tape.watch(model.trainable_weights)
	temp = model(inputs)

jacobian = tape.jacobian(temp, model.trainable_weights)
```

**Other info / logs**
It produces the following error messages.

*** Python 2.7.15 Error Output ***
```
Traceback (most recent call last):
  File "test.py", line 18, in <module>
    jacobian = tape.jacobian(temp, model.trainable_weights)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/eager/backprop.py", line 1121, in jacobian
    sys.exc_info()[2])
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/eager/backprop.py", line 1113, in jacobian
    parallel_iterations=parallel_iterations)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py", line 189, in pfor
    return f()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/eager/function.py", line 2362, in __call__
    graph_function, args, kwargs = self._maybe_define_function(args, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/eager/function.py", line 2703, in _maybe_define_function
    graph_function = self._create_graph_function(args, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/eager/function.py", line 2593, in _create_graph_function
    capture_by_value=self._capture_by_value),
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/framework/func_graph.py", line 978, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/framework/func_graph.py", line 968, in wrapper
    raise e.ag_error_metadata.to_exception(e)
ValueError: in converted code:

    /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py:183 f  *
        return _pfor_impl(loop_fn, iters, parallel_iterations=parallel_iterations)
    /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py:256 _pfor_impl
        outputs.append(converter.convert(loop_fn_output))
    /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/ops/parallel_for/pfor.py:1280 convert
        output = self._convert_helper(y)
    /Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/tensorflow_core/python/ops/parallel_for/pfor.py:1460 _convert_helper
        (y_op.type, y_op, converted_inputs))

    ValueError: No converter defined for LeakyReluGrad
    name: "loop_body/LeakyReluGrad"
    op: "LeakyReluGrad"
    input: "loop_body/MatMul"
    input: "loop_body/LeakyReluGrad/features"
    attr {
      key: "T"
      value {
        type: DT_FLOAT
      }
    }
    attr {
      key: "alpha"
      value {
        f: 0.300000011921
      }
    }
    
    inputs: [WrappedTensor(t=<tf.Tensor 'loop_body/MatMul/pfor/Reshape_1:0' shape=(100, 100, 2) dtype=float32>, is_stacked=True, is_sparse_stacked=False), WrappedTensor(t=<tf.Tensor 'loop_body/LeakyReluGrad/features:0' shape=(100, 2) dtype=float32>, is_stacked=False, is_sparse_stacked=False)]. 
    Either add a converter or set --op_conversion_fallback_to_while_loop=True, which may run slower

Encountered an exception while vectorizing the jacobian computation. Vectorization can be disabled by setting experimental_use_pfor to False.
```

*** Python 3.7.0 Error Output***
```
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/backprop.py", line 1113, in jacobian
    parallel_iterations=parallel_iterations)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py", line 189, in pfor
    return f()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/function.py", line 2362, in __call__
    graph_function, args, kwargs = self._maybe_define_function(args, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/function.py", line 2703, in _maybe_define_function
    graph_function = self._create_graph_function(args, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/function.py", line 2593, in _create_graph_function
    capture_by_value=self._capture_by_value),
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/framework/func_graph.py", line 978, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/framework/func_graph.py", line 968, in wrapper
    raise e.ag_error_metadata.to_exception(e)
ValueError: in converted code:

    /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py:183 f  *
        return _pfor_impl(loop_fn, iters, parallel_iterations=parallel_iterations)
    /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py:256 _pfor_impl
        outputs.append(converter.convert(loop_fn_output))
    /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/pfor.py:1280 convert
        output = self._convert_helper(y)
    /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/pfor.py:1460 _convert_helper
        (y_op.type, y_op, converted_inputs))

    ValueError: No converter defined for LeakyReluGrad
    name: "loop_body/LeakyReluGrad"
    op: "LeakyReluGrad"
    input: "loop_body/MatMul"
    input: "loop_body/LeakyReluGrad/features"
    attr {
      key: "T"
      value {
        type: DT_FLOAT
      }
    }
    attr {
      key: "alpha"
      value {
        f: 0.30000001192092896
      }
    }
    
    inputs: [WrappedTensor(t=<tf.Tensor 'loop_body/MatMul/pfor/Reshape_1:0' shape=(100, 100, 2) dtype=float32>, is_stacked=True, is_sparse_stacked=False), WrappedTensor(t=<tf.Tensor 'loop_body/LeakyReluGrad/features:0' shape=(100, 2) dtype=float32>, is_stacked=False, is_sparse_stacked=False)]. 
    Either add a converter or set --op_conversion_fallback_to_while_loop=True, which may run slower


During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "test.py", line 18, in <module>
    jacobian = tape.jacobian(temp, model.trainable_weights)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/backprop.py", line 1121, in jacobian
    sys.exc_info()[2])
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/six.py", line 702, in reraise
    raise value.with_traceback(tb)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/backprop.py", line 1113, in jacobian
    parallel_iterations=parallel_iterations)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py", line 189, in pfor
    return f()
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/function.py", line 2362, in __call__
    graph_function, args, kwargs = self._maybe_define_function(args, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/function.py", line 2703, in _maybe_define_function
    graph_function = self._create_graph_function(args, kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/eager/function.py", line 2593, in _create_graph_function
    capture_by_value=self._capture_by_value),
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/framework/func_graph.py", line 978, in func_graph_from_py_func
    func_outputs = python_func(*func_args, **func_kwargs)
  File "/Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/framework/func_graph.py", line 968, in wrapper
    raise e.ag_error_metadata.to_exception(e)
ValueError: in converted code:

    /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py:183 f  *
        return _pfor_impl(loop_fn, iters, parallel_iterations=parallel_iterations)
    /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/control_flow_ops.py:256 _pfor_impl
        outputs.append(converter.convert(loop_fn_output))
    /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/pfor.py:1280 convert
        output = self._convert_helper(y)
    /Library/Frameworks/Python.framework/Versions/3.7/lib/python3.7/site-packages/tensorflow_core/python/ops/parallel_for/pfor.py:1460 _convert_helper
        (y_op.type, y_op, converted_inputs))

    ValueError: No converter defined for LeakyReluGrad
    name: "loop_body/LeakyReluGrad"
    op: "LeakyReluGrad"
    input: "loop_body/MatMul"
    input: "loop_body/LeakyReluGrad/features"
    attr {
      key: "T"
      value {
        type: DT_FLOAT
      }
    }
    attr {
      key: "alpha"
      value {
        f: 0.30000001192092896
      }
    }
    
    inputs: [WrappedTensor(t=<tf.Tensor 'loop_body/MatMul/pfor/Reshape_1:0' shape=(100, 100, 2) dtype=float32>, is_stacked=True, is_sparse_stacked=False), WrappedTensor(t=<tf.Tensor 'loop_body/LeakyReluGrad/features:0' shape=(100, 2) dtype=float32>, is_stacked=False, is_sparse_stacked=False)]. 
    Either add a converter or set --op_conversion_fallback_to_while_loop=True, which may run slower

Encountered an exception while vectorizing the jacobian computation. Vectorization can be disabled by setting experimental_use_pfor to False.
```

**Comment by ravikyram** [comment]
@cianciosa 
I have tried on colab with TF version 2.1.0-rc2 .Please find the gist for [python3](https://colab.research.google.com/gist/ravikyram/2b9d8404b0c8e2b1969602e563db4dc0/untitled598.ipynb) and [python2](https://colab.research.google.com/gist/ravikyram/9b62bd4eaf07da2912ac14990f93a34a/untitled599.ipynb) here. Is this the expected behavior?. Thanks!

**Comment by cianciosa** [comment]
Since a LeakyReLU function has a well defined although discontinuous first derivative, I would have expected that computing the jacobian would proceed without throwing an exception.

If you comment out the LeakyReLU line and uncomment the normal ReLU line,
```
#model.add(tensorflow.keras.layers.LeakyReLU())
model.add(tensorflow.keras.layers.ReLU())
```
then you would see the behavior more a kin to what I would have expected.

**Comment by jvishnuvardhan** [comment]
@cianciosa I cannot reproduce the error when I ran your code with recent `tf-nightly`. [Here](https://colab.research.google.com/gist/jvishnuvardhan/2da0fa8376324480ef1b7abea655b48e/untitled61.ipynb) is the gist for your reference. Thanks!

Please close the issue if this was resolved for you. Thanks!

**Comment by cianciosa** [comment]
Verified it's fixed via tf_nightly-2.2.0.dev20200402.
